src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java

```diff
@@ -12,15 +12,15 @@ public boolean equals(String o) {
             return true;
         }
     }
-    public static class TestClass2 { // violation, no correct `equals`
+    public static class TestClass2 {
         public int hashCode() { // violation 'without .* of 'equals()'.'
             return 1;
         }
         public boolean equals(String o) {
             return true;
         }
     }
-    public static class TestClass3 { // violation, no `hashCode`
+    public static class TestClass3 {
         public boolean equals(Object o) { // violation 'without .* of 'hashCode()'.'
```

Comment:
`VIOLATION_WITH_EXPLANATION_PATTERN` has a comma while these don't, so the new regexp doesn't match these.

Why are we updating the regexp if it is not needed for these to match?

Reply:
it match https://github.com/checkstyle/checkstyle/blob/e071377c4d6e049491aef610b04bc106e8a8b6a3/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java#L80-L81

attentioon to `,?`

Reply:
Which is a different pattern and now brings up a separate question. Why do we have these similar patterns? We removed quotes from the other one so why are we keeping it in this one?

IMO, very similar patterns with no real reason on which to use will just confuse users. I am fine with this being a separate issue, but it should be an issue.

Reply:
They are separate.
With quotes is more strict as it demands quoted text to match violation message.

Without quotes is free text , whatever author think is reasonable to share that is not a message.